main/java/mjc/beans/PurchaseList.java

```diff
@@ -0,0 +1,112 @@
+package main.java.mjc.beans;
+
+import main.java.mjc.exceptions.CsvLineException;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.*;
+
+public class PurchaseList{
+    private List<AbstractPurchase> purchases;
+    private final Comparator<AbstractPurchase> comparator;
+    private boolean isSorted = false;
+
+    public PurchaseList(String csv, Comparator<AbstractPurchase> comparator) {
+        this.comparator = comparator;
+        try {
+            readFromFile(csv);
+        } catch (IOException e) {
+            purchases = new ArrayList<>();
+        }
+    }
+
+    private void readFromFile(String csv) throws IOException {
+        BufferedReader reader = new BufferedReader(new FileReader(csv));
+        String csvLine;
+        purchases = new ArrayList<>();
+
+        while ((csvLine = reader.readLine()) != null) {
+            try {
+                purchases.add(PurchaseFactory.getPurchase(csvLine));
+            } catch (CsvLineException e) {
+                System.err.println(csvLine);
+            }
+        }
+        reader.close();
+    }
+
+
+    public void insert(int index, AbstractPurchase purchase) {
+        if (index < 0) {
+            index = 0;
+        } else if (index > purchases.size()) {
+            index = purchases.size();
+        }
+        purchases.add(index, purchase);
+        isSorted = false;
+    }
+
+    public void delete(int from, int to) {
```

Comment:
```suggestion
    public int delete(int from, int to) {
```
it makes sense to return the number of deleted items.